src/lib/core/prbuf.c

```diff
@@ -369,3 +376,170 @@ prbuf_iterator_next(struct prbuf_iterator *iter, struct prbuf_entry *result)
 	result->ptr = iter->current->data;
 	return 0;
 }
+
+void
+prbuf_reader_close(struct prbuf_reader_iterator *iter)
+{
+	ibuf_destroy(&iter->buf);
+}
+
+/**
+ * Check whether buffer header is sane.
+ */
+static int
+prbuf_check_header(struct prbuf_header *header)
+{
+	if (header->version > prbuf_version) {
+		diag_set(IllegalParams,
+			 "unknown format version %d", header->version);
```

Comment:
`IllegalParams` doesn't look like a proper error to raise on invalid file format. I guess that's why the old implementation didn't set diag.

Reply:
Discussed f2f and decided to add new `FileFormatError`.